sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRowToColumnarExec.scala

```diff
@@ -757,19 +770,22 @@ object GeneratedUnsafeRowToCudfRowIterator extends Logging {
          |}
          |
          |final class SpecificUnsafeRowToColumnarBatchIterator extends ${classOf[UnsafeRowToColumnarBatchIterator].getName} {
+         |  private final org.apache.spark.sql.catalyst.expressions.UnsafeProjection unsafeProj;
          |
          |  ${ctx.declareMutableStates()}
          |
          |  public SpecificUnsafeRowToColumnarBatchIterator(Object[] references) {
-         |    super((scala.collection.Iterator<UnsafeRow>)references[0],
-         |      (org.apache.spark.sql.catalyst.expressions.Attribute[])references[1],
-         |      (com.nvidia.spark.rapids.CoalesceSizeGoal)references[2],
-         |      (com.nvidia.spark.rapids.GpuMetric)references[3],
-         |      (com.nvidia.spark.rapids.GpuMetric)references[4],
-         |      (com.nvidia.spark.rapids.GpuMetric)references[5],
-         |      (com.nvidia.spark.rapids.GpuMetric)references[6],
-         |      (com.nvidia.spark.rapids.GpuMetric)references[7],
-         |      (com.nvidia.spark.rapids.GpuMetric)references[8]);
+         |    super(
```

Comment:
`references` are no longer used in the body of this function. Should we remove them from the arguments?

Reply:
It is referenced.  Each of the *Ref values expands to a specific index in `references` with an explicit cast to the expected class.  It's producing the same string text to compile as it did before, but with the advantage that it doesn't need to manually keep the reference indices and `addReferenceObj` call order in sync.